pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -355,6 +354,61 @@ public SuccessResponse reloadSegment(
     }
   }
 
+  /**
+   * Resets the segment of the table, by disabling and then enabling it.
+   * This API will take segments to OFFLINE state, wait for External View to stabilize, and then back to ONLINE/CONSUMING state,
+   * thus effective in resetting segments or consumers in error states.
+   */
+  @POST
+  @Path("segments/{tableNameWithType}/{segmentName}/reset")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Resets a segment by first disabling it, waiting for external view to stabilize, and finally enabling it again",
+      notes = "Resets a segment by disabling and then enabling a segment")
+  public SuccessResponse resetSegment(
+      @ApiParam(value = "Name of the table with type", required = true) @PathParam("tableNameWithType") String tableNameWithType,
+      @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
+      @ApiParam(value = "Time in millis to wait for external view to converge") @QueryParam("externalViewWaitTimeMs") long externalViewWaitTimeMs) {
```

Comment:
All that this parameter does is to override the admin command wait time. Why not call it something like that? We can then add it to any admin command now or later with the same name.
Suggested:
```suggestion
      @ApiParam(value = "Maximum time in milliseconds to wait for reset to be completed") @QueryParam("maxWaitTimeMs") long externalViewWaitTimeMs) {
```

Reply:
This is not a generic max wait time, but a wait time for external view to stabilize.

Reply:
I am trying to avoid documenting what external view means. We know it as pinot devs, but probably better to minimize exposure of this to basic beginner users (and I think that is who we are targeting here). For all practical purposes, the ev wait will really translate to the total admin command wait time, so a re-use of that same term will save us some explanation -- is my thought.  I am flexible on this